index.bs

```diff
@@ -450,8 +450,27 @@ To <dfn>pullDatagrams</dfn>, given a {{WebTransport}} object |transport|, run th
   1. Set |datagrams|.{{[[IncomingDatagramsPullPromise]]}} to a new promise.
   1. Return |datagrams|.{{[[IncomingDatagramsPullPromise]]}}.
 1. Let |bytes| and |timestamp| be the result of [=dequeuing=] |queue|.
-1. Let |chunk| be a new {{Uint8Array}} object representing |bytes|.
-1. [=ReadableStream/Enqueue=] |chunk| to |transport|.{{[[Datagrams]]}}.{{WebTransportDatagramDuplexStream/[[Readable]]}}.
+1. Let |buffer|, |offset|, |maxBytes|, and |chunk| be null.
+1. If |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}'s
+   [=ReadableStream/current BYOB request view=] is not null, then:
+  1. Let |view| be |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}'s
+     [=ReadableStream/current BYOB request view=].
+  1. If |view|'s [=BufferSource/byte length=] is less than the size of |bytes|,
+     [=ReadableStream/error=] |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}
+     with a {{RangeError}}.
+  1. Otherwise:
+    1. Set |offset| to |view|.\[[ByteOffset]].
+    1. Set |maxBytes| to |view|'s [=BufferSource/byte length=].
+    1. Set |buffer| to |view|'s [=BufferSource/underlying buffer=].
+    1. Set |chunk| to |view|.
+1. Otherwise:
+  1. Set |offset| to 0.
+  1. Set |maxBytes| to the size of |bytes|.
+  1. Set |buffer| be a [=new=] {{ArrayBuffer}} with |maxBytes| size. If allocating the
+     {{ArrayBuffer}} fails, return [=a promise rejected with=] a {{RangeError}}.
+  1. Set |chunk| to a new {{Uint8Array}} object with |buffer| and |offset|.
+1. Write |bytes| into |buffer| with offset |offset|, up to |maxBytes| bytes.
+1. [=ReadableStream/Enqueue=] |chunk| to |datagrams|.{{WebTransportDatagramDuplexStream/[[Readable]]}}.
```

Comment:
But it might be surprising to always get Uint8Array if the reader receives Uint16Array for byob for example. The caller need to check the type of view anyway to properly retrieve non-byob chunks, so maybe not too much troublesome?

cc @domenic

Reply:
> What I'd like to happen is the if you pass a typed array with `BYTES_PER_ELEMENT > 1` to `read(view)` then it will error the stream. I'm not sure if the Streams standard exposes enough information to let us do that.

Should be possible as "current BYOB request view" gives a JS typed array object which should have that information. But the exception can only happen from the pull callback, not immediately from the read call. 🤔

Reply:
It's not clear to me from this conversation if https://github.com/whatwg/streams/pull/1263 suffices for your case. It'd be lovely if it did.

I plan to merge that PR as-is. If there are improvements we could make to it that would make it usable by both Fetch and WebTransport, we can fast-follow with such updates.

Reply:
> I plan to merge that PR as-is. If there are improvements we could make to it that would make it usable by both Fetch and WebTransport, we can fast-follow with such updates.

sgtm

Reply:
>It's not clear to me from this conversation if whatwg/streams#1263 suffices for your case. It'd be lovely if it did.

IMO it does, except the type and size of the BYOB view should be checked before calling it. That shouldn't be problematic.

Reply:
Can we have an ISSUE paragraph here at least?

Reply:
> IMO it does, except the type and size of the BYOB view should be checked before calling it.

I tried doing this and fixing the issue in the latest changes. Would you mind taking a look again, @saschanaz? Thank you!